.travis.yml

```diff
@@ -0,0 +1,55 @@
+language: go
+go:
+  - 1.13
+python:
+  - 3.4
+before_install:
+- |
+    git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '\.md$' || {
+      echo "Only docs were updated, stopping build process."
+      exit
+    }
+jobs:
+  include:
+  - name: "Check Header"
+    env: TEST=check_header
+  - name: "Build & Unit Test"
+    env: TEST=unit_test
+  - name: "Code Scan"
+    env: TEST=code_scan
+
+  
+script: >
+  if [ x${TEST} = xcheck_header ]; then
+    git clone --branch=devops https://github.com/vmware/singleton.git devops
+    cp $TRAVIS_BUILD_DIR/devops/check_headers.py .
+    chmod +x check_headers.py
+    python3 ./check_headers.py -f "$(git diff --name-only --diff-filter=d $TRAVIS_COMMIT_RANGE)"
+  fi &&
+  if [ x${TEST} = xunit_test ]; then
+    make build
+    ls -ltr builds
+    make test
```

Comment:
can you take a look at one failure in https://travis-ci.com/github/vm-mounika/singleton/jobs/498652991 line 1141 --- FAIL, is it a test case failure?

Reply:
Yes, It its. This is caused by win/linux difference. https://github.com/vmware/singleton/pull/1077 is to fix this.

Reply:
Why unit test stage still passed?

Reply:
@surite @Xiaochao8 I have moved build and test to separate steps.Now pipeline is failing with failing test ,Can you please check ?

Reply:
The pipeline is canceled. Do you know why?

Reply:
@Xiaochao8 I was testing in my branch it is running other branch as well so canceled it.